clients/vmdisplay/vmdisplay-server.cpp

```diff
@@ -290,11 +290,9 @@ int VMDisplayServer::init(int domid,
 
 int VMDisplayServer::cleanup()
 {
-	if (running) {
-		pthread_join(metadata_thread, NULL);
-		pthread_join(input_thread, NULL);
-		pthread_mutex_destroy(&mutex);
-	}
+	pthread_join(metadata_thread, NULL);
```

Comment:
I would rather leave if(running) in here. Instead, in SIGINT handler, we should set running to 1. That way cleanup will definitely go into this if condition.